cmd/microcloud/preseed.go

```diff
@@ -903,31 +903,39 @@ func (p *Preseed) Parse(s *service.Handler, c *initConfig, installedServices map
 		c.systems[peer] = system
 	}
 
-	allResources := map[string]*lxdAPI.Resources{}
-	for peer, system := range c.systems {
-		// Skip any systems that had direct configuration.
-		if len(system.MicroCephDisks) > 0 || len(system.TargetStoragePools) > 0 || len(system.StoragePools) > 0 {
-			continue
+	var checkFilterZFS, checkFilterCeph bool
+	for _, system := range p.Systems {
+		if checkFilterZFS && checkFilterCeph {
+			break
 		}
 
-		setupStorage := false
-		for _, cfg := range system.JoinConfig {
-			if cfg.Entity == "storage-pool" {
-				setupStorage = true
-				break
-			}
+		if (DirectStorage{} == system.Storage.Local) {
+			checkFilterZFS = true
 		}
 
-		if setupStorage {
-			continue
+		if len(system.Storage.Ceph) == 0 {
+			checkFilterCeph = true
 		}
+	}
 
-		cert := system.ServerInfo.Certificate
+	if !checkFilterCeph && len(p.Storage.Ceph) > 0 {
+		return nil, fmt.Errorf("Ceph disk filter cannot be used. All systems have explicitly specified disks")
```

Comment:
I think it's also worth putting a comment in the reference preseed file in the docs to indicate you can either explicitly or implicitly specific disks.